src/mod_roster.erl

```diff
@@ -226,18 +223,14 @@ hooks(Host) ->
      {roster_get_versioning_feature, Host, ?MODULE, get_versioning_feature, 50},
      {get_personal_data, Host, ?MODULE, get_personal_data, 50}].
 
-get_roster_entry(LUser, LServer, Jid) ->
```

Comment:
I though that backends, like for the case of ejabberd_sm, can stay with the binaries versions, as only this frontend (`mod_roster`) module is the one accessing the backends, so the backends can rely that the parameters given have already been stringprepped and need no further processing. Though it could be done in the future as well, perhaps for code quality, but I think it's less urgent when it comes to performance reasons.